modules/transport/PostMessageTransportBackend.js

```diff
@@ -0,0 +1,149 @@
+import Postis from 'postis';
+
+/**
+ * The list of methods of incomming postis messages that we have to support for
+ * backward compatability for the users that are directly sending messages to
+ * Jitsi Meet (without using external_api.js)
+ *
+ * @type {string[]}
+ */
+const legacyIncomingMethods = [ 'display-name', 'toggle-audio', 'toggle-video',
+    'toggle-film-strip', 'toggle-chat', 'toggle-contact-list',
+    'toggle-share-screen', 'video-hangup', 'email', 'avatar-url' ];
+
+/**
+ * The list of methods of outgoing postis messages that we have to support for
+ * backward compatability for the users that are directly listening to the
+ * postis messages send by Jitsi Meet(without using external_api.js).
+ *
+ * @type {string[]}
+ */
+const legacyOutgoingMethods = [ 'display-name-change', 'incoming-message',
+    'outgoing-message', 'participant-joined', 'participant-left',
+    'video-ready-to-close', 'video-conference-joined',
+    'video-conference-left' ];
+
+/**
+ * The postis method used for all messages.
+ *
+ * @type {string}
+ */
+const POSTIS_METHOD_NAME = 'data';
+
+/**
+ * The default options for postis.
+ *
+ * @type {Object}
+ */
+const defaultPostisOptions = {
+    window: window.opener || window.parent
+};
+
+/**
+ * Implements message transport using the postMessage API.
+ */
+export default class PostMessageTransportBackend {
+    /**
+     * Creates new PostMessageTransportBackend instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor(options = {}) {
+        const postisOptions = Object.assign({}, defaultPostisOptions,
+            options.postisOptions);
+
+        this.postis = Postis(postisOptions);
+
+        this._enableLegacyFormat = options.enableLegacyFormat;
+
+        if (!this._enableLegacyFormat) {
```

Comment:
This sounds like if the legacy format is not enabled then we go and look at legacyIncomingMethods, which is wrong. So I guess the name of the variable isn't very clear.